packages/embark/src/lib/modules/tests/test.js

```diff
@@ -306,21 +289,34 @@ class Test {
           next(null, accounts);
         });
       },
-      function deploy(accounts, next) {
-        self.events.request('deploy:contracts:test', () => {
+      function waitForProvidersReady(accounts, next) {
+        self.events.request('console:provider:ready', () => {
           next(null, accounts);
         });
       },
       function getWeb3Object(accounts, next) {
         self.events.request('blockchain:get', (web3) => {
-          next(null, accounts, web3);
+          // global web3 used in the tests, not in the vm
+          global.web3 = web3;
```

Comment:
You do this both here and in `resetEmbarkJS`. Maybe we can get rid of one  of those

Reply:
I'd say, you could rename `resetEmbarkJS` to `resetSandbox` that resets embarkJs and the VM at the same time

Reply:
Nice look again, thanks. Have removed one. 

In regards to renaming, I left the event name as `runcode:embarkjs:reset`, as all the other event names pertaining to the initialization code use `embarkjs`.